jOOQ-meta/src/main/java/org/jooq/meta/xml/XMLRoutineDefinition.java

```diff
@@ -167,4 +169,16 @@ protected void init0() {
             }
         }
     }
+
+    private Name getOverloadRoutineNamePart() {
+        Name qualifiedName = getQualifiedNamePart();
+        String lastName = qualifiedName.last();
+        if (lastName == null) {
+            return qualifiedName;
+        } else {
+            Name[] previousPath = Arrays.copyOf(qualifiedName.parts(), qualifiedName.parts().length - 1);
+            String overloadLastName = lastName + (isBlank(getOverload()) ? "" : "_" + getOverload());
```

Comment:
I think it may well be a coincidence that our generated overload index is the same as the suffix of the specific name information. Running this query here in my own PostgreSQL installation:

```sql
select specific_name
from information_schema.routines
where routine_name = 'pg_advisory_xact_lock'
```

Yields:

```
specific_name             |
--------------------------|
pg_advisory_xact_lock_3093|
pg_advisory_xact_lock_3089|
```

Rather than using this suffix (which is only used for disambiguation by jOOQ), we should use the "specific name" information, explicitly.

I'll devise a fix.

Reply:
In any case, the equality check you've fixed is definitely where the bug happens

Reply:
Ah, I see why you've used this logic. It's the logic taken from `XMLGenerator`. But that's just one way to implement the "specific name", we shouldn't rely on it.